scipy/stats/mstats_basic.py

```diff
@@ -362,7 +362,9 @@ def pearsonr(x,y):
         prob = 0.
     else:
         t_squared = (df / ((1.0 - r) * (1.0 + r))) * r * r
-        prob = betai(0.5*df, 0.5, df/(df + t_squared))
+        tmp = df / (df + t_squared)
+        tmp = np.where(tmp < 1.0, tmp, 1.0)
```

Comment:
These repeated lines indicate that ``betai`` is providing something of use here. So instead of adding the same boilerplate before every call to ``special.betainc``, it would be better to rename ``betai`` to ``_betai`` and keep using it internally here.